core/src/main/java/org/elasticsearch/index/query/QueryBuilders.java

```diff
@@ -47,6 +47,13 @@ public static MatchAllQueryBuilder matchAllQuery() {
     }
 
     /**
+     * A query that match no documents.
+     */
+    public static MatchNoneQueryBuilder matchNoDocsQuery() {
```

Comment:
maybe call the method `matchNoneQuery` along the same lines as `matchAllQuery` ?